src/tree/IngressResource.ts

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Ingress } from "@azure/arm-appcontainers";
+import { IActionContext, TreeItemIconPath } from "@microsoft/vscode-azext-utils";
+import { ThemeIcon } from "vscode";
+import { IngressConstants } from "../constants";
+import { ContainerAppChildResource } from "../resolver/ContainerAppChildResource";
+import { ContainerAppExtResource } from "../resolver/ContainerAppExtResource";
+import { ContainerAppResource } from "../resolver/ContainerAppResource";
+import { ContainerAppsExtResourceBase } from "../resolver/ContainerAppsExtResourceBase";
+import { localize } from "../utils/localize";
+import { treeUtils } from "../utils/treeUtils";
+
+const label: string = localize('ingress', 'Ingress');
+
+export class IngressResource extends ContainerAppChildResource<Ingress | undefined> {
+    public data: Ingress;
+
+    public label: string;
+    public description: string | undefined;
+    public idSuffix: string = 'ingress';
+
+    constructor(data: Ingress | undefined, containerApp: ContainerAppResource, parentId: string) {
+        super(containerApp);
+        this.data = data ?? {};
+        this.id = `${parentId}/${this.idSuffix}`
+
+        this.label = label;
+        this.description = data ? undefined : localize('disabled', 'Disabled');
+
+        this.isParent = !!data;
```

Comment:
I was thinking some about the quick pick logic, and I was thinking--should we instead do `isLeaf`? i.e. is it more interesting whether something actually has children, or whether it's even possible for it to have children?

Reply:
I'm pretty impartial to either approach. I just went with `isParent` because if I leave it undefined, it'll default to `false`. My thought was that leaf nodes are more common than parents, but I feel like both convey the same information if you just ! it